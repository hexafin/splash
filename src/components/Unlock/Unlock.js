import React, { Component } from "react"
import {
	View,
	Text,
	StyleSheet,
	Image,
	TouchableOpacity,
	Animated
} from "react-native"
import { colors } from "../../lib/colors"
import { defaults, icons } from "../../lib/styles"
import Keypad from '../universal/Keypad'
import ReactNativeHapticFeedback from 'react-native-haptic-feedback';
import LinearGradient from 'react-native-linear-gradient';
import TouchID from "react-native-touch-id"


class Unlock extends Component {
	constructor(props) {
		super(props)
		this.state = {
			value: '',
		}
		this.onChangeText = this.onChangeText.bind(this)
		this.shakeAnimation = new Animated.Value(0)
	}

	componentDidMount() {
		TouchID.authenticate("Login with biometric").then(success => {
			if (success) {
				this.props.navigation.navigate('SwipeApp')
			}
		})
	}

	onChangeText(text) {
		if (text.length <= 4) {
			this.setState({value: text})
			ReactNativeHapticFeedback.trigger("impactLight", true)

			// if correct passcode TODO: use real passcode
			if (text == '1234') {
				this.props.navigation.navigate('SwipeApp')
			} else if (text.length == 4) {
				// if incorrect passcode
				Animated.spring(this.shakeAnimation, {
					toValue: 1,
					useNativeDriver: true,
				}).start(() => {
					this.shakeAnimation.setValue(0)
					this.setState({value: ''})
				})
				ReactNativeHapticFeedback.trigger("impactHeavy", true)
			}
		}
	}

	render() {
		const dropIcon = (index) => (this.state.value.length >= index) ? 'whiteDrop' : 'purpleDrop'
		const shakeTransform =  {
			transform: [
				{
					translateX: this.shakeAnimation.interpolate({
						inputRange: [0, 0.25, 0.5, 0.75, 1],
						outputRange: [0, 40, 0, -40, 0]
					})
				}
			]
		}

		return (
			<LinearGradient colors={['#5759D5', '#4E50E6']} style={styles.container}>
				<Image source={icons.whiteSplash} style={styles.splashLogo} resizeMode={'contain'}/>
				<Animated.View style={[styles.drops, shakeTransform]}>
					<Image source={icons[dropIcon(1)]} style={styles.drop} resizeMode={'contain'}/>
					<Image source={icons[dropIcon(2)]} style={styles.drop} resizeMode={'contain'}/>
					<Image source={icons[dropIcon(3)]} style={styles.drop} resizeMode={'contain'}/>
					<Image source={icons[dropIcon(4)]} style={styles.drop} resizeMode={'contain'}/>
				</Animated.View>
				<Keypad primaryColor={'#484AD4'}
						pressColor={'#6466F6'}
						textColor={'#FFFFF'}
						onChange={(text) => this.onChangeText(text)}
						disabled={(this.state.value.length >= 4) ? true : false}
						decimal={false}
						value={this.state.value}
						arrow={'white'} />
				{/*<TouchableOpacity onPress={() => console.log('forgot')}>
					<Text style={styles.forgotText}>Forgot?</Text>
				</TouchableOpacity>*/}
			</LinearGradient>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		paddingTop: 75,
	},
	splashLogo: {
		height: 34,
		width: 26,
		marginBottom: 76,
		alignSelf: 'center'
	},
	drops: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		marginBottom: 113,
		paddingHorizontal: 105,
	},
	drop: {
		height: 15,
		width: 12,
	},
	forgotText: {
		fontSize: 14,
		color: colors.white,
		paddingTop: 38,
		alignSelf: 'center'
	},
})

export default Unlock
