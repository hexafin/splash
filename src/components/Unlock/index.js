import Unlock from "./Unlock"
import {connect} from "react-redux"
import {bindActionCreators} from "redux"

const mapStateToProps = (state) => {
    return {
      splashtag: state.user.entity.username || "yourname"
    }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({

    }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Unlock)
