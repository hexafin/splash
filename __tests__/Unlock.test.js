import "react-native";
import React from "react";
import Unlock from "../src/components/Unlock/Unlock";

it("Unlock renders correctly", () => {
	const snap = shallow(
		<Unlock
			navigation={{
				state: {
					params: {}
				}
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});
