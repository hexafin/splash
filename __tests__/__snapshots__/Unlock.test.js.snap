// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unlock renders correctly 1`] = `
<LinearGradient
  colors={
    Array [
      "#5759D5",
      "#4E50E6",
    ]
  }
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
      "paddingTop": 75,
      "paddingVertical": 40,
      "position": "relative",
      "width": 750,
    }
  }
>
  <Component>
    <Component
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/whiteSplash.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "height": 34,
          "width": 26,
        }
      }
    />
  </Component>
  <AnimatedComponent
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingHorizontal": 105,
          "paddingTop": 50,
          "width": 750,
        },
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <Component
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/purpleDrop.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 12,
        }
      }
    />
    <Component
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/purpleDrop.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 12,
        }
      }
    />
    <Component
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/purpleDrop.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 12,
        }
      }
    />
    <Component
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/icons/purpleDrop.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 12,
        }
      }
    />
  </AnimatedComponent>
  <Keypad
    arrow="white"
    decimal={false}
    delete={true}
    disabled={false}
    onChange={[Function]}
    pressColor="#6466F6"
    primaryColor="#484AD4"
    style={
      Object {
        "bottom": 40,
        "position": "absolute",
        "width": 750,
      }
    }
    textColor="white"
  />
</LinearGradient>
`;
